File: trip-backend/src/models/password_reset.py
"""PasswordReset model (password_resets table)"""

from sqlalchemy import Column, String, DateTime, Boolean
from sqlalchemy.orm import relationship

from src.models.base import Base, BaseModelWithoutUpdatedAt


class PasswordReset(Base, BaseModelWithoutUpdatedAt):
    """PasswordReset model (no updated_at field)"""
    
    __tablename__ = "password_resets"
    __table_args__ = ({"comment": "密码重置表"},)
    
    # Fields (id and created_at inherited from BaseModelWithoutUpdatedAt)
    email = Column(
        String(100),
        nullable=False,
        comment="用户邮箱"
    )
    token = Column(
        String(255),
        unique=True,
        nullable=False,
        comment="重置令牌"
    )
    expires_at = Column(
        DateTime(timezone=True),
        nullable=False,
        comment="过期时间"
    )
    used = Column(
        Boolean,
        default=False,
        nullable=False,
        comment="是否已使用"
    )
    
    def __repr__(self):
        return f"<PasswordReset(id={self.id}, email={self.email})>"

File: trip-backend-java/src/main/java/com/trip/backend/domain/entity/PasswordReset.java
package com.trip.backend.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;

/**
 * 密码重置实体（对应 Python models/password_reset.py）
 * - 表结构：email, token(unique), expires_at, used, created_at
 */
@Entity
@Table(name = "password_resets", indexes = {
    @Index(name = "idx_password_resets_token", columnList = "token", unique = true),
    @Index(name = "idx_password_resets_email", columnList = "email")
})
@Getter
@Setter
public class PasswordReset {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 100)
    private String email;

    @Column(nullable = false, unique = true, length = 255)
    private String token;

    @Column(nullable = false)
    private OffsetDateTime expiresAt;

    @Column(nullable = false)
    private Boolean used = false;

    @Column(nullable = false)
    private OffsetDateTime createdAt;

    protected PasswordReset() {}

    public PasswordReset(String email, String token, OffsetDateTime expiresAt) {
        this.email = email;
        this.token = token;
        this.expiresAt = expiresAt;
        this.createdAt = OffsetDateTime.now();
        this.used = false;
    }

    /**
     * 检查令牌是否有效（未过期且未使用）
     */
    public boolean isValid() {
        return !used && expiresAt.isAfter(OffsetDateTime.now());
    }
}
